src/simulation/ApplyLoad.cpp

```diff
@@ -0,0 +1,266 @@
+
+    mLoadCodeKey = contractCodeLedgerKey;
+
+    auto const& lm = mApp.getLedgerManager();
+    auto uploadTx = mTxGenerator.createUploadWasmTransaction(
+        lm.getLastClosedLedgerNum() + 1, 0, wasmBytes, contractCodeLedgerKey,
+        std::nullopt);
+
+    closeLedger({uploadTx.second});
+
+    for (auto kvp : mTxGenerator.getAccounts())
+    {
+        auto salt = sha256("Load contract " + std::to_string(kvp.first));
+
+        auto createTx = mTxGenerator.createContractTransaction(
+            lm.getLastClosedLedgerNum() + 1, 0, contractCodeLedgerKey,
+            wasmBytes.size() + 160, salt, std::nullopt);
+        closeLedger({createTx.second});
+
+        auto instanceKey =
+            createTx.second->sorobanResources().footprint.readWrite.back();
+
+        TxGenerator::ContractInstance instance;
+        instance.readOnlyKeys.emplace_back(mLoadCodeKey);
+        instance.readOnlyKeys.emplace_back(instanceKey);
+        instance.contractID = instanceKey.contractData().contract;
+        mLoadInstances.emplace(kvp.first, instance);
+    }
+}
+
+void
+ApplyLoad::benchmark()
+{
+    auto& lm = mApp.getLedgerManager();
+    std::vector<TransactionFrameBasePtr> txs;
+
+    auto resources = multiplyByDouble(lm.maxLedgerResources(true),
+                                      TRANSACTION_QUEUE_SIZE_MULTIPLIER);
+
+    auto const& accounts = mTxGenerator.getAccounts();
+    std::vector<uint64_t> shuffledAccounts(accounts.size());
+    std::iota(shuffledAccounts.begin(), shuffledAccounts.end(), 0);
+    stellar::shuffle(std::begin(shuffledAccounts), std::end(shuffledAccounts),
+                     gRandomEngine);
+
+    for (auto accountIndex : shuffledAccounts)
+    {
+        auto it = accounts.find(accountIndex);
+        releaseAssert(it != accounts.end());
+
+        auto instanceIter = mLoadInstances.find(it->first);
+        releaseAssert(instanceIter != mLoadInstances.end());
+        auto const& instance = instanceIter->second;
+        auto tx = mTxGenerator.invokeSorobanLoadTransaction(
+            lm.getLastClosedLedgerNum() + 1, it->first, instance,
+            rust_bridge::get_write_bytes().data.size() + 160, std::nullopt);
+
+        if (!anyGreater(tx.second->getResources(false), resources))
+        {
+            resources -= tx.second->getResources(false);
+        }
+        else
+        {
+            for (size_t i = static_cast<size_t>(Resource::Type::OPERATIONS);
+                 i <= static_cast<size_t>(Resource::Type::WRITE_LEDGER_ENTRIES);
+                 ++i)
+            {
+                auto type = static_cast<Resource::Type>(i);
+                if (tx.second->getResources(false).getVal(type) >
+                    resources.getVal(type))
+                {
+                    CLOG_INFO(Perf, "Ledger {} limit hit during tx generation",
+                              Resource::getStringFromType(type));
+                }
+            }
+            break;
+        }
```

Comment:
Does this `else` branch indicate a misconfiguration? If so maybe this should function should return `false` or something indicating that `benchmark` failed. Then `runApplyLoad` in `CommandLine.cpp` can check the result and do something (error and stop?) with it. Currently `runApplyLoad` just moves on to reporting stats about the run and it might be easy to miss this `INFO` message indicating that `benchmark` failed to generate transactions.

Reply:
That's not an issue as we're expected to hit a certain ledger limit eventually. But what we should actually check is that we did indeed hit this condition, or else we might specify arbitrarily large ledger limits while being in fact limited by the number of accounts used.

Reply:
I added an assert to make sure we hit one of the limits. I also remove the numAccount argument, and instead set it to `ledgerMaxTxCount * SOROBAN_TRANSACTION_QUEUE_SIZE_MULTIPLIER + 1`.